src/java.base/unix/classes/sun/nio/fs/UnixPath.java

```diff
@@ -857,13 +860,76 @@ public Path toRealPath(LinkOption... options) throws IOException {
             result = result.resolve(element);
         }
 
-        // check file exists (without following links)
+        // check whether file exists (without following links)
         try {
             UnixFileAttributes.get(result, false);
         } catch (UnixException x) {
             x.rethrowAsIOException(result);
         }
-        return result;
+
+        // Return if the file system is not both case insensitive and retentive
+        if (!fs.isCaseInsensitiveAndPreserving())
+            return result;
+
+        UnixPath path = fs.rootDirectory();
+
+        SecurityManager sm = System.getSecurityManager();
```

Comment:
I assume this can be removed too.

Reply:
Yes, in c193b3b48f56e2c581efaf212f2770c9add36a0a.